pkg/api/v1alpha1/cluster.go

```diff
@@ -201,6 +202,7 @@ var clusterConfigValidations = []func(*Cluster) error{
 	validateCPUpgradeRolloutStrategy,
 	validateControlPlaneLabels,
 	validatePackageControllerConfiguration,
+	validateCloudstackK8sVersion,
```

Comment:
Are these validations suppose to be aware of provider context?  Seems like this validation belongs more in the cloudstack package.

Reply:
We would like to have this validation in webhook so it could be fast. Having the method in cloudstack creates import cycle